src/main/kotlin/io/github/nomisrev/repo/UserPersistence.kt

```diff
@@ -124,6 +128,19 @@ fun userPersistence(
       ensureNotNull(info) { UserNotFound("userId=$userId") }
     }
 
+    override suspend fun unfollowProfile(
+      followedUsername: String,
+      followerId: UserId
+    ): Either<DomainError, Unit> = either {
+      usersQueries.transaction {
+        usersQueries.selectByUsername(followedUsername).executeAsOneOrNull()?.let {
+          usersQueries.selectSecurityByEmail(it.email).executeAsOne().let {
+            followingQueries.delete(it.id.serial, followerId.serial)
+          }
+        }
+      }
+    }
```

Comment:
I wrote a new `followingQueries.delete` query, that simplifies this use-case.

```sql
delete:
DELETE FROM following
WHERE followed_id=(SELECT id FROM users WHERE users.username=:username) AND follower_id = :followerId;
```

This should change the code to:

```suggestion
    ): Unit =
      followingQueries.delete(followedUsername, followerId.serial)
```

Reply:
Committing suggestion in [9a6eba6](https://github.com/nomisRev/ktor-arrow-example/pull/190/commits/9a6eba68b857727e087a4d2930354c149f563dc9)